testhelpers/core.go

```diff
@@ -184,3 +187,158 @@ func RequireCreateStorages(ctx context.Context, t *testing.T) (state.Tree, vm.St
+func NewFakeVMContext(message *types.Message, state interface{}) *FakeVMContext {
+	randomness := MakeRandomBytes(32)
+	addressGetter := address.NewForTestGetter()
+	return &FakeVMContext{
+		TestMessage:        message,
+		TestStorage:        &testStorage{state: state},
+		TestBlockHeight:    types.NewBlockHeight(0),
+		TestBalance:        types.ZeroAttoFIL,
+		TestRandomness:     randomness,
+		TestIsAccountActor: true,
+		Charger: func(cost types.GasUnits) error {
+			return nil
+		},
+		Sampler: func(sampleHeight *types.BlockHeight) ([]byte, error) {
+			return randomness, nil
+		},
+		Sender: func(to address.Address, method string, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error) {
+			return [][]byte{}, 0, nil
+		},
+		Addresser: func() (address.Address, error) {
+			return addressGetter(), nil
+		},
+		ActorCreator: func(addr address.Address, code cid.Cid, initalizationParams interface{}) error {
+			return nil
+		},
+	}
+}
+
+// Message is the message that triggered this invocation
+func (tc *FakeVMContext) Message() *types.Message {
+	return tc.TestMessage
+}
+
+// Storage provides and interface to actor state
+func (tc *FakeVMContext) Storage() exec.Storage {
+	return tc.TestStorage
+}
+
+// Send sends a message to another actor
+func (tc *FakeVMContext) Send(to address.Address, method string, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error) {
+	return tc.Sender(to, method, value, params)
+}
+
+// AddressForNewActor creates an address to be used to create a new actor
+func (tc *FakeVMContext) AddressForNewActor() (address.Address, error) {
+	return tc.Addresser()
+}
+
+// BlockHeight is the current chain height
+func (tc *FakeVMContext) BlockHeight() *types.BlockHeight {
+	return tc.TestBlockHeight
+}
+
+// MyBalance is the balance of the current actor
+func (tc *FakeVMContext) MyBalance() types.AttoFIL {
+	return tc.TestBalance
+}
+
+// IsFromAccountActor returns true if the actor that sent the message is an account actor
+func (tc *FakeVMContext) IsFromAccountActor() bool {
+	return tc.TestIsAccountActor
+}
+
+// Charge charges gas for the current action
+func (tc *FakeVMContext) Charge(cost types.GasUnits) error {
+	return tc.Charger(cost)
+}
+
+// SampleChainRandomness provides random bytes used in verification challenges
+func (tc *FakeVMContext) SampleChainRandomness(sampleHeight *types.BlockHeight) ([]byte, error) {
+	return tc.Sampler(sampleHeight)
+}
+
+// CreateNewActor creates an actor of a given type
+func (tc *FakeVMContext) CreateNewActor(addr address.Address, code cid.Cid, initalizationParams interface{}) error {
+	return tc.ActorCreator(addr, code, initalizationParams)
+}
+
+// Verifier provides an interface to the proofs verifier
+func (tc *FakeVMContext) Verifier() exec.Verifier {
+	return tc.TestVerifier
+}
+
+// FakeVerifier is a fake implementation of a proof verifier
+type FakeVerifier struct {
+	Valid       bool
+	Err         error
+	SealRequest verification.VerifySealRequest
```

Comment:
Please document that the first two determine behaviour, the latter two are captured from callers. Maybe call this `LastSealRequest` or `ReceivedSealRequest` or similar.